src/BifrostBridge.php

```diff
@@ -12,69 +12,25 @@
 
 class BifrostBridge
 {
-    /** @var callable|string|null */
-    protected static $userClassResolver = null;
+    use Traits\ResolvesRoleClass, Traits\ResolvesUserClass, Traits\ResolvesUser;
 
-    /** @var callable|string|null */
-    protected static $roleClassResolver = null;
-
-    /** @var callable|string|null */
-    protected static $userWithoutEmailResolver = null;
-
-    /**
-     * @param callable|string|null $callback
-     * @return void
-     */
-    public static function resolveUserClassUsing($callback): void
-    {
-        static::$userClassResolver = $callback;
-    }
-
-    /**
-     * @param callable|string|null $callback
-     * @return void
-     */
-    public static function resolveRoleClassUsing($callback): void
-    {
-        static::$roleClassResolver = $callback;
-    }
-
-    /**
-     * @param callable|string|null $callback
-     * @return void
-     */
-    public static function resolveUserWithoutEmailUsing($callback): void
-    {
-        static::$userWithoutEmailResolver = $callback;
-    }
-
-    public static function retrieveUserWithoutEmail(BifrostUserData $data): ?Model
+    // Resolvers
+    public static function resolveAndUpdateUser(BifrostUserData $data): ?Model
```

Comment:
`\Illuminate\Contracts\Auth\Authenticatable` ipv `Model, dat is een vereiste voor `Auth::login($user)`